apps/labs/posts/scipy-array-api.md

```diff
@@ -0,0 +1,441 @@
+array's namespace using `xp.asarray` after it.
+This will raise exceptions for arrays on a different device to the CPU, as justified in
+[the RFC](https://github.com/scipy/scipy/issues/18286) (tldr: we want to avoid silent device transfers).
+
+Here is an example from `scipy.fft`'s Discrete Sine and Cosine Transforms, where `_pocketfft` is the backend
+which SciPy uses for NumPy arrays:
+
+```python
+def _execute(pocketfft_func, x, type, s, axes, norm, 
+             overwrite_x, workers, orthogonalize):
+    xp = array_namespace(x)
+    x = np.asarray(x)
+    y = pocketfft_func(x, type, s, axes, norm,
+                       overwrite_x=overwrite_x, workers=workers,
+                       orthogonalize=orthogonalize)
+    return xp.asarray(y)
+
+def dctn(x, type=2, s=None, axes=None, norm=None,
+         overwrite_x=False, workers=None, *, orthogonalize=None):
+    return _execute(_pocketfft.dctn, x, type, s, axes, norm, 
+                    overwrite_x, workers, orthogonalize)
+```
+
+As a side note, there has also been work to support the calling of some specific backends instead of hitting
+compiled code in `scipy.special` (in [this PR](https://github.com/scipy/scipy/pull/19023/)),
+which enables GPU support for at least some provider libraries.
+I will keep this blog post focused on just array-agnostic code since this work in `special` should ultimately be
+replaced by a more robust dispatching mechanism, whereas the array-agnostic code is here to stay!
+
+However, as mentioned above, `fft` is a standard extension module, so the standard can still help us with some
+compiled code!
+For the functions in the extension, we can check whether the array's namespace implements the extension with
+`if hasattr(xp, 'fft')`, and use `xp.fft` if so.
+Since NumPy has an `fft` submodule, we need to be careful to still use SciPy's implementation for NumPy arrays,
+for which we have the `is_numpy` helper.
+
+Here is how that looks for `scipy.fft.fft`:
+
+```python
+def _execute_1D(func_str, pocketfft_func, x, n, axis, norm, overwrite_x, workers, plan):
+    xp = array_namespace(x)
+
+    if is_numpy(xp):
+        return pocketfft_func(x, n=n, axis=axis, norm=norm,
+                              overwrite_x=overwrite_x, workers=workers, plan=plan)
+
+    norm = _validate_fft_args(workers, plan, norm)
+    if hasattr(xp, 'fft'):
+        xp_func = getattr(xp.fft, func_str)
+        return xp_func(x, n=n, axis=axis, norm=norm)
+
+    x = np.asarray(x)
+    y = pocketfft_func(x, n=n, axis=axis, norm=norm)
+    return xp.asarray(y)
+
+def fft(x, n=None, axis=-1, norm=None,
+        overwrite_x=False, workers=None, *, plan=None):
+    return _execute_1D('fft', _pocketfft.fft, x, n=n, axis=axis, norm=norm,
+                       overwrite_x=overwrite_x, workers=workers, plan=plan)
+```
+
+After converting the production functions, we also need to convert the tests!
+
+We have the `@array_api_compatible` decorator, which parametrizes `xp` with the
```

Comment:
Can we have another level?

If yes, consider adding a summary at the beginning of the section with the steps needed, then go through them like you did.